### DISCUSSION TITLE
[backend] Add Paperclip to automatically generate OpenAPI v2 specifications for the REST API

### DISCUSSION THREAD

**Discussion by TobiasDeBruijn** [torrust/torrust-index-backend#232] [Feature requests]
Basically as the title says.

The advantage of this would be that an OpenApi V2 specification can be automatically generated from the code. This would be helpful for those implementing client libraries for Torrust. It should be quite trivial to add too, I'd be more than happy to PR this.

Thanks.

**Comment by mickvandijke** [comment]
We would certainly appreciate a PR to implement Paperclip :)

**Comment by TobiasDeBruijn** [comment]
Awesome.

I'll first need to port Paperclip to support Actix v4, after that it *should* be easy for Torrust.

**Comment by josecelano** [comment]
Is [this package](https://github.com/paperclip-rs/paperclip) the one you would use? We have migrated the backend to Axum and I do not know is it has support for it. And it seems [it's not maintained anymore](https://github.com/paperclip-rs/paperclip/issues/308). I'm going to transfer the issue to the backend repo discussion and make it a "feature request" discussion.

**Comment by josecelano** [comment]
Relates to: https://medium.com/netwo/documenting-api-for-actix-web-b575adb841a1

**Comment by josecelano** [comment]
I think we first should discuss whether we need this and if we need it, research about alternatives.
